src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,75 @@
+import React from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<number>;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === totalPages;
+
+  const handlePageChange = (page: number) => {
+    if (page !== currentPage && onPageChange) {
+      onPageChange(page);
+    }
```

Comment:
The check for `onPageChange` being truthy is unnecessary because it's not optional in your Props interface. You can safely call `onPageChange(page)` without checking if it exists.